web/modules/custom/bnf/src/Services/BnfImporter.php

```diff
@@ -264,101 +67,37 @@ public function loadData(string $uuid, string $endpointUrl, string $nodeType = '
-        'query' => $query,
-      ],
-      // Make sure that the server is HTTPS.
-      'verify' => TRUE,
-    ]);
-
-    $data = json_decode($response->getBody()->getContents(), TRUE);
-    $nodeData = $data['data'] ?? NULL;
-
-    if (empty($nodeData)) {
-      $this->logger->error(
-        'Could not find any node data in GraphQL response. @query',
-        ['@query' => $query]
-      );
-
-      throw new \Exception('Could not retrieve content values.');
-    }
-    return $nodeData;
-
-  }
-
-  /**
-   * Importing a node from a GraphQL source endpoint.
-   */
-  public function importNode(string $uuid, string $endpointUrl, string $nodeType = 'article'): NodeInterface {
-    $nodeStorage = $this->entityTypeManager->getStorage('node');
     try {
-      $data = $this->loadData($uuid, $endpointUrl, $nodeType);
-      // Find node data, e.g. nodeArticle, nodePage etc.
-      $nodeDataKeys = preg_grep('/^node[A-Z]/', array_keys($data));
-      $nodeDataKey = $nodeDataKeys ? reset($nodeDataKeys) : [];
-      $nodeData = $data[$nodeDataKey] ?? [];
+      $endpointConfig = new SailorEndpointConfig(MangleUrl::server($endpointUrl));
+      Configuration::setEndpointFor(GetNode::class, $endpointConfig);
+      $response = GetNode::execute($uuid);
 
-      $baseUrl = $data['info']['url'] ?? NULL;
-      $nodePath = $nodeData['path'] ?? NULL;
-      $canonicalUrl = $nodeData['canonicalUrl']['url'] ?? NULL;
-      $canonicalTitle = $nodeData['canonicalUrl']['title'] ?? NULL;
+      $nodeData = $response->data?->node;
 
-      // If no custom canonical URL is set, we'll set the context of where
-      // we have exported the data from.
-      if (empty($canonicalUrl) && $baseUrl && $nodePath) {
-        $canonicalUrl = "$baseUrl$nodePath";
-        $canonicalTitle = $data['info']['name'] ?? NULL;
+      if (!$nodeData) {
+        throw new \RuntimeException('Could not fetch content.');
       }
 
-      $nodeCreation = [
-        'status' => NodeInterface::NOT_PUBLISHED,
-        'type' => $nodeType,
-        'uuid' => $uuid,
-        'title' => $nodeData['title'],
-        'field_paragraphs' => $this->getParagraphs($nodeData),
-        'field_canonical_url' => [
-          'uri' => $canonicalUrl,
-          'title' => $canonicalTitle,
-        ],
-      ];
-
-      /** @var \Drupal\node\NodeInterface $node */
-      $node = $nodeStorage->create($nodeCreation);
-      $node->save();
+      $node = $this->mapperManager->map($nodeData);
 
       $node->set(BnfStateEnum::FIELD_NAME, BnfStateEnum::Imported->value);
+
+      $node->set('status', NodeInterface::NOT_PUBLISHED);
+
       $node->save();
     }
-    catch (\Exception $e) {
+    catch (\Throwable $e) {
       $this->logger->error(
-        'Failed to create node data. @message',
+        'Failed to import content. @message',
         ['@message' => $e->getMessage()]
       );
 
-      throw new \Exception('Could not save content.');
+      throw new \RuntimeException('Could not import content.');
```

Comment:
Consider why we both log *and* rethrow here.

As I see it we only need to do one of the two. If we have handed the error we can log it but not rethrow. If not we can just rethrow and let the outer code decide.

Otherwise we end up with multiple logs of the same error.

Reply:
Well, yeah, I obviously didn't think much about this part.